probe/pkg/runtime/runtime.go

```
@@ -45,19 +43,32 @@ func (r *Runtime) RunLoop(ctx context.Context) error {
 }
 
 // RunSingle executes a single probe run.
-func (r *Runtime) RunSingle(ctx context.Context) error {
-	ctxTimeout, cancel := context.WithTimeout(ctx, r.Config.RuntimeRunTimeout)
+func (r *Runtime) RunSingle(ctx context.Context) (errReturn error) {
+	probeRunCtx, cancel := context.WithTimeout(ctx, r.Config.ProbeRunTimeout)
 	defer cancel()
-	defer r.CleanUp()
+	defer func() {
+		cleanupCtx, cancel := context.WithTimeout(context.Background(), r.Config.ProbeCleanUpTimeout)
+		defer cancel()
+		cleanupDone := concurrency.NewSignal()
+
+		go r.scheduleCleanup(cleanupCtx, &cleanupDone)
+
+		select {
+		case <-cleanupCtx.Done():
+			errReturn = cleanupCtx.Err()
```

I haven't seen the usage of named return parameters in combination with go routine as such here - will this actually lead to the `context cancelled / context deadline exceeded` error being returned from `RunSingle` here?

Either way, probably nice to have a comment describing this here, it wasn't not totally obvious to me at first sight.

---

The return is not assigned from the goroutine, but just from a deferred function. So yeah it should return the context error. However on second look, I this might not be such a good idea in case `Execute` already returned an error, because then we are overwriting that. So I think just logging the clean up error would be better.